### DISCUSSION TITLE
Caret position hook

### DISCUSSION THREAD

**Discussion by atumas-bananamilk** [mantinedev/mantine#4326] [Feature requests]
Is it possible to get a hook to return the live position of a caret (cursor) within Input / Textarea fields?

This would help a lot with emoji pickers to insert an emoji at cursor instead of appending it at the end all the time.

**Comment by rtivital** [comment]
I do not think that it is a very common use case. You can easily do that with one line of code in `onKeyUp` – https://codesandbox.io/s/wonderful-fire-1c6g6g?file=/src/App.tsx

**Reply by atumas-bananamilk** [reply]
Doesn't work if using EmojiPicker (adding text programmatically).

**Comment by atumas-bananamilk** [comment]
For anyone reading this - `onKeyUp` doesn't work because some text can be added programmatically (e.g. EmojiPicker).
Solved by:
`const position = ref?.current?.selectionStart`
